File: Ros2/cpp_pkg/src/numberpublisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::chrono_literals;



class numberpublisher: public rclcpp::Node
{   
    int count = 0;

    public:
           numberpublisher(): Node("numberpublisher")
           {
                 
                 publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10 );
                 timer_ = this->create_wall_timer(1s,std::bind(&numberpublisher::publishdata,this));

                 RCLCPP_INFO(get_logger(),"Data start publishing.....");

           }
  

    private:
           
           void publishdata()
           {
            auto msg= example_interfaces::msg::Int64();
            count = count+ 1;
            msg.data= this->count;
            publisher_ -> publish(msg);

           }

rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
rclcpp::TimerBase::SharedPtr timer_;


};




int main(int argc, char **argv)
{

    rclcpp::init(argc,argv);
    auto node = std::make_shared<numberpublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;



}

File: Ros2/Python_pkg/pycode/numberpublisher.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64

class numberpublisher(Node):
    def __init__(self):
        super().__init__("numberpublisher")
        self.publisher = self.create_publisher(Int64,"number",10)
        self.timer = self.create_timer(1,self.publisher_station)
        self.get_logger().info("Number start publishing....")
        self.count = 0


    def publisher_station(self):
        msg = Int64()
        self.count  = self.count + 1
        msg.data = self.count
        self.publisher.publish(msg)
        self.get_logger().info("Publish Number :"+str(msg.data))


def main(args=None):
    rclpy.init(args=args)
    node = numberpublisher()
    rclpy.spin(node)
    rclpy.shutdown()



if "__init__"== "__main__":
    main()